pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -334,16 +335,28 @@ def _load_config(self, config):
         return config
 
     def setup_distributed(self):
-        super().setup_distributed()
+        reset_seed()
+
+        # determine which process we are and world size
+        self.set_world_ranks()
+
+        # set warning rank
+        rank_zero_only.rank = self.global_rank
+
+        self.init_deepspeed_distributed()
+
+        # set the ranks and devices
+        self.dist.rank = self.global_rank
+        self.dist.device = self.root_device
         if not self._config_initialized:
             self._format_config()
             self._config_initialized = True
 
-    def init_ddp_connection(self, global_rank: Optional[int] = None, world_size: Optional[int] = None) -> None:
+    def init_deepspeed_distributed(self) -> None:
         if platform.system() != "Windows":
```

Comment:
Does DeepSpeed work well with Windows?

Reply:
Good question, I think it does support it however I haven't tested this. It would be nice to find a user that has :)